shells/pipes-shell/context.js

```diff
@@ -0,0 +1,77 @@
+import {now} from '../../build/platform/date-web.js';
+//import {generateId} from '../../modalities/dom/components/generate-id.js';
+import {RamSlotComposer} from '../lib/ram-slot-composer.js';
+import {Utils} from '../lib/utils.js';
+import {ArcHost} from '../lib/arc-host.js';
+import {Stores} from '../lib/stores.js';
+import {Schemas} from './schemas.js';
+
+export const Context = class {
+  constructor(storage) {
+    this.initContext(storage);
+  }
+  async initContext(storage) {
+    this.context = await Utils.parse('');
+    await this.initAddressStore(this.context);
+    await this.initPipesArc(storage);
+  }
+  async initAddressStore(context) {
+    const store = await Stores.create(context, {
+      name: 'pipe-entities',
+      id: 'pipe-entities',
+      schema: Schemas.PipeEntity,
+      isCollection: true,
+      tags: null,
+      storageKey: null
+    });
+    this.contextEntityStore = store;
+    //console.log(store);
+  }
+  async initPipesArc(storage) {
+    console.log('Context::initPipesArc');
+    const id = 'pipes-arc';
+    const manifest = `import 'https://$particles/Pipes/BackgroundPipes.recipes'`;
+    const composer = new RamSlotComposer();
+    const host = new ArcHost(null, storage, composer);
+    this.pipesArc = await host.spawn({id, manifest});
+    // TODO(sjmiles): findById would be better,
+    // but I can't get the id to materialize via manifest
+    this.entityStore = this.pipesArc._stores[0];
+    if (this.entityStore) {
+      await this.entityStoreChange(await this.getInitialChange(this.entityStore));
+      this.entityStore.on('change', info => this.entityStoreChange(info), this);
+    } else {
+      console.log('Context::initPipesArc: failed to find entityStore');
+    }
+    dumpStores([this.entityStore]);
+  }
+  async getInitialChange(store) {
+    const change = {add: []};
+    const values = await store.toList();
+    values.forEach(value => change.add.push({value}));
+    return change;
+  }
+  async entityStoreChange(change) {
+    console.log(change);
```

Comment:
Could be overly verbose logging?

Reply:
Hard to say, logging becomes helpful when problems occur. Until we have some successful roundtrips on actual device, I don't know what kind of information we will need. I tried to limit myself to critical junctures; this spot relies on a runtime callback to propagate observedEntities into the context.